lib/livebook/settings.ex

```diff
@@ -182,20 +182,26 @@ defmodule Livebook.Settings do
     attributes = env_var |> Map.from_struct() |> Map.to_list()
 
     with :ok <- storage().insert(:env_vars, env_var.key, attributes),
-         :ok <- broadcast_env_vars_change() do
+         :ok <- broadcast_env_vars_change({:env_var_set, env_var}) do
       {:ok, env_var}
     end
   end
 
   @doc """
-  Deletes an environment variable from given id.
+  Unsets an environment variable from given id.
 
-  Also, it notifies interested processes about environment variables data change.
+  With success, notifies interested processes about environment variable
+  deletion. Otherwise, it does nothing.
   """
-  @spec delete_env_var(String.t()) :: :ok
-  def delete_env_var(id) do
-    storage().delete(:env_vars, id)
-    broadcast_env_vars_change()
+  @spec unset_env_var(String.t()) :: :ok
+  def unset_env_var(id) do
+    if env_var_exists?(id) do
+      env_var = fetch_env_var!(id)
+      storage().delete(:env_vars, id)
+      broadcast_env_vars_change({:env_var_unset, env_var})
```

Comment:
You could also do this:

```suggestion
      broadcast_env_vars_change({:env_var_set, %{env_var | value: nil}})
```

This way you don't need `System.delete_env`, you only need a single message, and you can rely on `System.put_env(envs)` as it deletes nil values since Elixir v1.14.0. :) WDYT?